File: backend/app/models/portfolio.py
from datetime import datetime


def new_portfolio_item(
    title: str,
    image_key: str,
    thumbnail_key: str,
    category: str = "general",
    description: str | None = None,
    sort_order: int = 0,
) -> dict:
    return {
        "title": title,
        "description": description,
        "category": category,
        "image_key": image_key,
        "thumbnail_key": thumbnail_key,
        "sort_order": sort_order,
        "is_visible": True,
        "created_at": datetime.utcnow(),
    }

File: src/models/portfolio.ts
export function newPortfolioItem(
  title: string,
  imageKey: string,
  thumbnailKey: string,
  category = 'general',
  description: string | null = null,
  sortOrder = 0,
  mediaId: string | null = null,
) {
  return {
    title,
    description,
    category,
    image_key: imageKey,
    thumbnail_key: thumbnailKey,
    sort_order: sortOrder,
    is_visible: true,
    media_id: mediaId,
    created_at: new Date(),
  };
}
